### ISSUE TITLE
Retrospective sessions should allow DateTime entry

### ISSUE THREAD

**Issue by gdevenyi** [DouglasNeuroInformatics/OpenDataCapture#1029]
Our native datatype is a DateTime object, we should allow retrospective DateTime specification.

My suspicion is the API call supports DateTime but the frontend doesn't expose a time component.

This is useful for MouseODC (and probably human stuff) because the time-of-day is a proxy for hormonal variation.

**Comment by david-roper** [comment]
Proposed Solution
Allow for an optional input box or selected field to be revealed (hour and minutes) when the user selects a retrospective session in the form. Thus allowing them to put the time of day if needed.

**Comment by gdevenyi** [comment]
> Allow for an optional input box or selected field to be revealed (hour and minutes) when the user selects a retrospective session in the form. Thus allowing them to put the time of day if needed.

Yes, except I believe the existing data structures are using a single DateTime like object for the Date entry right now, so merging the two entries together might be tricky. I was more thinking about a single input function which handles both.

**Comment by joshunrau** [comment]
Yes, it is indeed possible to input the exact time of day, but there is no UI component for this at the moment. David's solution would work, but it is a bit cumbersome. @david-roper could you try implementing a clock component in libui?
